src/libexpr/primops.cc

```diff
@@ -115,7 +115,7 @@ StringMap EvalState::realiseContext(const NixStringContext & context)
     return res;
 }
 
-static SourcePath realisePath(EvalState & state, const PosIdx pos, Value & v, bool resolveSymlinks = true)
+static SourcePath realisePath(EvalState & state, const PosIdx pos, Value & v, std::optional<SymlinkResolution> resolveSymlinks = SymlinkResolution::Full)
```

Comment:
BTW, instead of an `std::optional`, wouldn't it be cleaner to have a `SymlinkResolution::None` alternative?

Reply:
Let's see how much of that footgun remains at all after working on the nicer fix.